README.md

```diff
@@ -151,28 +149,71 @@ constants must be common to all clients and the server:
-build copy the above client files to the device. Edit `local.py` as described
-below and copy it to the device. Ensure the device has a stored WiFi connection
-and run te demo.
+build copy the above client files to the device. Edit `local.py` for each example
+as described below and copy it to the device. Ensure the device has a stored 
+WiFi connection and run the demo.
+To run the demo the file `local.py` for the corresponding example should be edited 
+for the server IP address.
+The demo supports up to four clients. Each client's `local.py` should be edited
+to give each client a unique client ID in range 1..4. Note that the ESP8266
+must have a stored network connection to access the server.
 
-Alternatively to maximise free RAM firmware can be built from source, freezing
+On the server ensure that `local.py` is on the path and run `s_app_cp.py`.
+
+Alternatively to maximise free RAM, firmware can be built from source, freezing
 `uasyncio`, `client.py` and `primitives.py` as bytecode.
 
 If a daily build is used it will be necessary to
 [cross compile](https://github.com/micropython/micropython/tree/master/mpy-cross)
 `client.py`
 
-To run the demo the file `local.py` should be edited for the server IP address.
-The demo supports up to four clients. Each client's `local.py` should be edited
-to give each client a unique client ID in range 1..4. Note that the ESP8266
-must have a stored network connection to access the server.
+To get the files onto your ESP8266 do NOT copy single files as this repository is
+built to be used as a python package. This means that you have to retain the file
+structure for it to work.
+The easiest way is to clone the repository:
+```
+git clone https://github.com/peterhinch/micropython-iot micropython_iot
+```
+It's important to clone it into a directory *micropyhton_iot* as python does not like
+packages that have a "-" character in their name. 
+Then you need [rshell](https://github.com/dhylands/rshell) and follow these commands:
+```
+rshell -p /dev/ttyS3  # adapt the port to your situation
+mkdir /pyboard/micropython_iot   # create directory on your esp822 
+(yes you need to use /pyboard here, at least I do)
+rsync micropython_iot /pyboard/micropython_iot -v -m
+(this can take a while, you might have to do it again if it fails early during the image
+transfer of "block_diagram_orig.odg". There's no exclude option for rsync yet)
+```
+Now you have successfully synchronized the repository onto your device.
 
-On the server ensure that `local.py` is on the path and run `s_app_cp.py`.
+The other way is to freeze it into the firmware by copying the repository to the
+micropython/ports/esp8266 directory.
+
+## 3.2 Usage
+
+On the esp8266 you can now run every example using the following syntax:
+```
+from micropython_iot.examples import c_app
+from micropython_iot.example_remote_control import c_comms_tx
+from micropython_iot.example_remote_control import c_comms_rx
+from micropython_iot.qos import c_qos
```

Comment:
A little unusal way of starting a python file on unix for the average user but works perfectly.